code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -336,6 +336,28 @@
 		remove_from_storage(I, T, 1)
 	finish_bulk_removal()
 
+/obj/item/storage/MouseDrop_T(atom/dropping, mob/living/user)
+	if(isliving(dropping))
+		scoop_inside(dropping, user)
+		return
+	..()
+
+/obj/item/storage/proc/scoop_inside(mob/living/scooped, mob/living/user)
+	if(!scooped.holder_type || scooped.buckled || scooped.pinned.len || scooped.mob_size > MOB_SIZE_SMALL || scooped != user || src.loc == scooped)
+		return
+
+	if(!do_after(user, 1 SECOND, src))
```

Comment:
Need to check scoopee loc, Adjacency, incapacitated status, etc. to make sure the bag or mob haven't been moved.